File: 0096-unique-binary-search-trees/0096-unique-binary-search-trees.cs
public class Solution {
    public int NumTrees(int n) {
        if(n==1)
        {
            return 1;
        }
        int[] dp = new int[n+1];
        for(int i=0; i<=n; i++)
        {
            dp[i] = -1;
        }
        dp[0] = 1;
        dp[1] = 1;
        dp[2] = 2;
        for(int i=3; i<=n; i++)
        {
            int s = 0;
            for(int k=1 ;k<=i; k++)
            {
                s += dp[k-1]*dp[i-k];
            }
            dp[i] = s;
        }
        return dp[n];
    }
}

File: 0096-unique-binary-search-trees/0096-unique-binary-search-trees.java
class Solution {
    public int numTrees(int n) {
        int[] countOfTrees = new int[n+1];
        countOfTrees[0] = 1;
        for(int i=1; i<=n; i++)
        {
            int current = 0;
            for(int leftSubTreeSize = 0; leftSubTreeSize < i; leftSubTreeSize++)
            {
                current += countOfTrees[leftSubTreeSize]*countOfTrees[i-leftSubTreeSize-1];
            }
            countOfTrees[i] = current;
        }
        return countOfTrees[n];
    }
}